packages/vscode-extension/src/telemetry/extTelemetry.ts

```diff
@@ -186,6 +193,34 @@ export namespace ExtTelemetry {
     reporter.sendTelemetryException(error, properties, measurements);
   }
 
+  export async function cacheTelemetryEventAsync(
+    eventName: string,
+    properties?: { [p: string]: string }
+  ) {
+    const telemetryEvents = {
+      eventName: eventName,
+      properties: {
+        [TelemetryProperty.CorrelationId]: lastCorrelationId,
```

Comment:
Is this correlation id related to deactivation event?

Reply:
Yes, it helps to connect the deactivate event with last operation. Otherwise, the last operation has no "end" event.